pkg/cloud/services/interfaces.go

```diff
@@ -65,22 +66,29 @@ type EC2Interface interface {
 	TerminateInstanceAndWait(instanceID string) error
 	DetachSecurityGroupsFromNetworkInterface(groups []string, interfaceID string) error
 
-	ReconcileLaunchTemplate(scope scope.LaunchTemplateScope, canUpdateLaunchTemplate func() (bool, error), runPostLaunchTemplateUpdateOperation func() error) error
 	ReconcileTags(scope scope.LaunchTemplateScope, resourceServicesToUpdate []scope.ResourceServiceToUpdate) error
 
 	DiscoverLaunchTemplateAMI(scope scope.LaunchTemplateScope) (*string, error)
-	GetLaunchTemplate(id string) (lt *expinfrav1.AWSLaunchTemplate, userDataHash string, err error)
+	GetLaunchTemplate(id string) (lt *expinfrav1.AWSLaunchTemplate, userDataHash string, userDataSecretKey *types.NamespacedName, err error)
 	GetLaunchTemplateID(id string) (string, error)
 	GetLaunchTemplateLatestVersion(id string) (string, error)
-	CreateLaunchTemplate(scope scope.LaunchTemplateScope, imageID *string, userData []byte) (string, error)
-	CreateLaunchTemplateVersion(id string, scope scope.LaunchTemplateScope, imageID *string, userData []byte) error
+	CreateLaunchTemplate(scope scope.LaunchTemplateScope, imageID *string, userDataSecretKey types.NamespacedName, userData []byte) (string, error)
+	CreateLaunchTemplateVersion(id string, scope scope.LaunchTemplateScope, imageID *string, userDataSecretKey types.NamespacedName, userData []byte) error
 	PruneLaunchTemplateVersions(id string) error
 	DeleteLaunchTemplate(id string) error
 	LaunchTemplateNeedsUpdate(scope scope.LaunchTemplateScope, incoming *expinfrav1.AWSLaunchTemplate, existing *expinfrav1.AWSLaunchTemplate) (bool, error)
 	DeleteBastion() error
 	ReconcileBastion() error
 }
 
+// ReconcileInterface encapsulates high-level reconciliation functions regarding EC2 reconciliation. It is separate
+// from EC2Interface so that we can test the behavior of our non-mock implementations. For example, by not mocking
+// the ReconcileLaunchTemplate function, but mocking EC2Interface, we can test which EC2 API operations would have
+// been called.
+type ReconcileInterface interface {
```

Comment:
`EC2Interface` mocks the AWS EC2 API, so `ReconcileLaunchTemplate` (which I moved out) and `ReconcileTags` (out of scope for my PR) should not be in there. If I move the `ReconcileTags` function as well into the new interface, it should become clearer for someone working on the code, particularly if we state that it's for the functionality of the controller as opposed to raw AWS requests. Given that change, what about `MachinePoolReconcileInterface` since those functions are exclusively used in `awsmachinepool_controller.go` and `awsmanagedmachinepool_controller.go`?

Reply:
I think that would clear up my confusion, sgtm.

Reply:
Done. Also fixed lint errors.